### PR TITLE
Resnet50 cleanups

### PR DESCRIPTION
**Pull request by steven-johnson**
Some cleanups to smooth the way for adding autotuning:
-- remove unused generator and cmake file
-- parameterize load_weights and process to be more configurable from the command line as to files, directories, etc
-- pytorch weights are now downloaded as part of the make process and stored in a bin subfolder, so 'make test' should work as-is (assuming pytorch is installed)
-- misc cleanup in process.cpp, including ability make loading nonpreprocessed images work.

### CODE REVIEW

File: apps/resnet_50/process.cpp [hunk 1]
```diff
@@ -217,22 +271,47 @@ std::vector<FnPtr> blockFns = {resnet50block0,
...
+  Buffer<float> image;
+  if (extension(input_file) == "bin") {
+    // load image that is prenormalized and transposed
+    //  (TODO: image size is hardcoded for panda)
+    image = buffer_from_file(input_file, {3, 224, 224});
+    std::cout << "Loading prenormalized, transposed image from " << input_file << std::endl;
+  } else {
+    Buffer<float> im_in = load_and_convert_image(input_file);
+    std::cout << "Loading image from " << input_file << std::endl;
+    assert(im_in.dimensions() == 3);
```
[the remark this answers is not in the document]
Comment (steven-johnson) [reply apps/resnet_50/process.cpp]: Interesting; what in the later code relies on this specific size? (Are the weights specific to that size?)

[the remark this answers is not in the document]
Comment (KarimaMa) [reply apps/resnet_50/process.cpp]: the resnet network expects the input to be specifically that size to work

[the remark this answers is not in the document]
Comment (steven-johnson) [reply apps/resnet_50/process.cpp]: Ouch. Did not realize. Will fix.

[the remark this answers is not in the document]
Comment (steven-johnson) [reply apps/resnet_50/process.cpp]: Done

**Comment by steven-johnson** [comment]
Merged as part of https://github.com/halide/Halide/pull/3563

**Pull request merged**